### PR TITLE
moves compliance row to ts source and fix bugs discovered by typing c…

### PR DESCRIPTION
**Pull request by theseyi**
…omponent interface and methods

### CODE REVIEW

File: wherehows-web/app/components/dataset-compliance-row.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+  /**
+   * The field identifier attribute
+   * @type {ComputedProperty<string>}
+   * @memberof DatasetComplianceRow
+   */
+  identifierField: ComputedProperty<string> = alias('field.identifierField');
+
+  /**
+   * The field's dataType attribute
```
Comment (igbopie) [inline wherehows-web/app/components/dataset-compliance-row.ts]: Are those automated comments?

File: wherehows-web/tests/integration/components/disable-bubble-input-test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+test('it invokes the click action on click', function(assert) {
+  const inputClickAction = () => {
+    assert.ok(true, 'disable input click action invoked');
+  };
+  this.set('inputClickAction', inputClickAction);
+
+  this.render(hbs`{{disable-bubble-input click=inputClickAction}}`);
+
+  triggerEvent('input[type=text]', 'click');
```
Comment (igbopie) [inline wherehows-web/tests/integration/components/disable-bubble-input-test.js]: You can use https://api.qunitjs.com/assert/expect to make sure that if the action is not called it fails.

Comment (theseyi) [reply wherehows-web/tests/integration/components/disable-bubble-input-test.js]: yea, forgot to add that here, good catch

File: wherehows-web/app/components/dataset-compliance-row.ts (hunk 1 above)
Comment (theseyi) [reply wherehows-web/app/components/dataset-compliance-row.ts]: not really, manually generated with document this in vscode, then modified
https://marketplace.visualstudio.com/items?itemName=joelday.docthis

**Pull request merged**